pyoptsparse/pyIPOPT/meson.build

```diff
@@ -0,0 +1,58 @@
+fs = import('fs')
+
+if get_option('ipopt_dir') != '' or fs.is_dir('Ipopt')
```

Comment:
In addition to the `IPOPT_DIR` approach, the original `setup.py` had several other ways of detecting the library:
- searching for a directory under `pyIPOPT`
- allowing for specifying include and lib dirs separately via `IPOPT_INC` and `IPOPT_LIB`

Reply:
Ugh... another annoying meson particularity. It finds the IPOPT library automatically, but then you can't get that path so you cant find the header...

Reply:
I'm actually going to try something real quick. I have an idea for how to handle this

Reply:
I'm realizing I misread your comment... searching `pyIPOPT` directory was already supported. I did push up a commit that cleaned up the logic a bit though. `IPOPT_INC` and `IPOPT_LIB` are not however, I can add that back in soon, but is there a use case where the headers and shared library don't follow the layout `$IPOPT_DIR/coin{,-or}` and `$IPOPT_DIR/lib`?

Reply:
If the user installs IPOPT via yum on CentOS for example, I think the system lib dir is typically `/usr/lib64/` instead of `/usr/lib/`. We can of course add in some sort of smart detection, but allowing the user to specify both independently may be a more flexible solution.

Reply:
Actually, I see that you already handle lib64 here. Maybe that's okay then, if we find a use case we can always add this feature back.

Reply:
Is it okay to resolve this?

Reply:
Yep!